internal/plugins/golang/v2/api.go

```diff
@@ -57,19 +58,26 @@ func (p *createAPISubcommand) Run() error {
 
 	// Find the new resource. Here we shouldn't worry about checking if one was found,
 	// since downstream plugins will do so.
-	var newResource config.GVK
-	for _, r := range p.config.Resources {
-		if _, hasResource := oldResources[r]; !hasResource {
-			newResource = r
-			break
+	newResources, err := p.config.GetResources()
+	if err != nil {
+		return err
+	}
+	var newResource resource.Resource
+	for _, newR := range newResources {
+		newResource = newR
+		for _, oldR := range oldResources {
+			if !oldR.GVK.IsEqualTo(newR.GVK) {
+				newResource = newR
+				break
```

Comment:
If you assign it outside of the conditional and it doesn't find any new resource, it will think that the last resource was the new resource:
```suggestion
		for _, oldR := range oldResources {
			if !oldR.GVK.IsEqualTo(newR.GVK) {
				newResource = newR
				break
```

Reply:
It is not true. This covers the scenario when it is the first resource of the project.